falcon/routing/static.py

```diff
@@ -83,6 +95,9 @@ def __call__(self, req, resp):
         if '..' in file_path or not file_path.startswith(self._directory):
             raise falcon.HTTPNotFound()  # pragma: nocover
 
+        if self._default_filename is not None and not os.path.isfile(file_path):
```

Comment:
PERF: `open()` already checks whether the file exists. Therefore, you can avoid an extra file system call by attempting to open the fallback file in the `except IOError` block, rather than calling `os.path.isfile()` in advance (filesystem calls are expensive relative to a few extra CPU cycles spent on exception handling).